tests/test_browser.py

```diff
@@ -4981,3 +4981,19 @@ def test_pthread_modularize_export_name(self):
                args=['-s', 'INITIAL_MEMORY=64MB', '-s', 'USE_PTHREADS=1',
                      '-s', 'PTHREAD_POOL_SIZE=8', '-s', 'MODULARIZE_INSTANCE=1',
                      '-s', 'EXPORT_NAME=MyModule', '--shell-file', 'shell.html'])
+
+  # Tests that it is possible to hook into/override a symbol defined in a system library.
+  @requires_graphics_hardware
+  def test_override_system_js_lib_symbol(self):
+    # This test verifies it is possible to override a symbol from WebGL library.
+
+    # When WebGL is implicitly linked in, the implicit linking should happen before any user --js-libraries, so that they can adjust
+    # the behavior afterwards.
+    self.btest(path_from_root('tests', 'test_override_system_js_lib_symbol.c'),
+               expected='5121',
+               args=['--js-library', path_from_root('tests', 'test_override_system_js_lib_symbol.js')])
+
+    # When WebGL is explicitly linked to in strict mode, the linking order on command line should enable overriding.
+    self.btest(path_from_root('tests', 'test_override_system_js_lib_symbol.c'),
+               expected='5121',
+               args=['-s', 'STRICT=1', '-lwebgl.js', '--js-library', path_from_root('tests', 'test_override_system_js_lib_symbol.js')])
```

Comment:
Why not just make this a regular non-browser test an just override something simple from library.js (e.g. chroot).?

Reply:
Or at least some JS library that doesn't require a browser test?  All it would need to do is assert the system library came first and then that injected symbol was the one that got used?

Reply:
Because I want to test my use case specifically, and the `-s STRICT=1` and `-lwebgl.js` have specific behavior here.

Reply:
I see.  It looks its really about `AUTO_JS_LIBRARIES=0` which is enabled when `STRICT=1`, right?  It looks like any of the following libraries would do:

```
        'library_webgl.js',                                                         
        'library_openal.js',                                                        
        'library_sdl.js',                                                                             
        'library_glut.js',                                                          
        'library_xlib.js',                                                          
        'library_egl.js',                                                           
        'library_glfw.js',                                                          
        'library_uuid.js',                                                          
        'library_glew.js',                                                          
        'library_idbstore.js',                                                      
        'library_async.js'   
```

I see a lot of the are browser related.   I guess async might work?   I know it s pain to re-write the test, so I won't push it, but a browser tests for this kind of thing does seem like overkill.

Hopefully we can remove AUTO_JS_LIBRARIES completely one day and this test would no longer be relevant I guess?

Actually seeing as this test is about AUTO_JS_LIBRARIES specifically can you change the the setting to be about that in particular rather than the more general STRICT=1?  lgtm with that change.